cabal-install/Distribution/Client/Install.hs

```diff
@@ -1439,7 +1436,7 @@ installUnpackedPackage verbosity installLock numJobs
                                       NoMultiInstance
                                       packageDBs ipkg'
 
-            return (Right (BuildResult docsResult testsResult ipkgs'))
+            return (Right (BuildResult docsResult testsResult (find ((==uid).installedUnitId) ipkgs')))
```

Comment:
As far as I can tell, in the legacy code path the stored `Maybe InstalledPackageInfo` is not actually used for anything. So, it is the case that we do register all of the convenience libs, but then we only keep the registration of the library in the build result. This doesn't seem to cause problems.